archetypal/template/umi_base.py

```diff
@@ -437,6 +457,57 @@ def get_unique(self):
 
         return obj
 
+    @property
+    def Parents(self):
+        """ Get the parents of an UmiBase Object"""
+        parents = {}
+        for (umi_class, lookup_children_of) in self._POSSIBLE_PARENTS:
+            possible_parents = UmiBase._CREATED_OBJECTS[umi_class + "s"]
+            for parent in possible_parents:
+                # For each parent, find the possible children which
+                # may equal self
+                keyed_children = []  # stores list of (key, child)
+                if callable(lookup_children_of):
+                    keyed_children = lookup_children_of(parent)
+                elif isinstance(lookup_children_of, list):
+                    keyed_children = [
+                        (attr, getattr(parent, attr))
+                        for attr in lookup_children_of
+                        if getattr(parent, attr, None)
+                    ]
+                else:
+                    raise ValueError(
+                        "The provided parent lookup is not valid.  It must be a list of str or return a list of enumerated children"
+                    )
+                for key, child in keyed_children:
+                    if child.id == self.id:
+                        if child == self:
+                            if parent not in parents:
+                                parents[parent] = set()
+                            parents[parent].add(key)
+        return parents
+
+    @property
+    def ParentTemplates(self):
+        """ Get the parent templates of an UmiBase object"""
+        templates = set()
+        for parent in self.Parents.keys():
+            # Recursive call terminates at Parent Template level, or if self.Parents is empty
+            templates = templates.union(parent.ParentTemplates)
+        return templates
+
+    @classmethod
+    def CREATED_OBJECTS(cls, object_class):
+        """ Return all objects of a given class that UmiBase is aware of
+            Args:
+                object_class (class): Inherits UmiBase
+        """
+        return cls._CREATED_OBJECTS[object_class.__name__ + "s"]
+
+    @property
+    def CREATED_OBJECTS(self):
```

Comment:
Could this be a private property instead of a public one? Trying to keep it clean as much as possible